src/main/org/deidentifier/arx/aggregates/StatisticsClassification.java

```diff
@@ -361,9 +363,105 @@ private static ClassificationMethod getClassifier(WrappedBoolean interrupt,
         double[] zerorConfidences = new double[numSamples * ( 1 + numClasses)];
         int confidencesIndex = 0;
                 
-        // For each fold as a validation set
-        for (int evaluationFold = 0; evaluationFold < folds.size(); evaluationFold++) {
-            
+        if (config.getEvaluateWithKfold()) {        
```

Comment:
The two versions should be put into two different methods, e.g.

evaluateWithCrossValidation(...)
evaluateWithTrainingSet(...)

Currently it is (a) hard to review, (b) there are some redundancies. For example, you are calculating the folds in line 351, even if you dont need them because you are using a training and test set.